src/main/java/duke/main/Storage.java

```diff
@@ -0,0 +1,111 @@
+package duke.main;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+import duke.task.Deadlines;
+import duke.task.Events;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todos;
+
+/**
+ * A class handling the saving and load the taskList
+ * into a file.
+ */
+public class Storage {
+
+    /**
+     * The path of the file where the taskList is saved
+     */
+    protected String filePath;
+
+    /**
+     * The constructor for the Storage
+     */
+    public Storage() {
+        String home = System.getProperty("user.home");
+        Path path = Paths.get(home, "data", "duke.txt");
+        File file = new File(path.toString());
+        try {
+            file.getParentFile().mkdirs();
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+            this.filePath = path.toString();
+        } catch (IOException e) {
+            System.out.println("Cannot create file!");
+        }
+    }
+
+    /**
+     * Writes the data into the file.
+     * @param data The content to be written into the file
+     */
+    public void writeData(String data) {
+        try {
+            FileWriter file = new FileWriter(this.filePath, false);
+            file.write(data);
+            file.close();
+        } catch (IOException e) {
+            //Throw DukeException
+        }
+
```

Comment:
I think you missed this empty line, as well as on line 79 and 110, can remove them!